src/visitors.rs

```diff
@@ -235,6 +248,33 @@ impl<'a, 'b: 'a, 'tcx: 'b> MirVisitor<'a, 'b, 'tcx> {
         self.summary_cache.set_summary_for(self.def_id, summary);
     }
 
+    /// Use the visitor to compute the state corresponding to promoted constants.
+    fn promote_constants(&mut self, mut state_with_parameters: Environment) -> Environment {
+        let saved_mir = self.mir;
+        let result_root = Path::LocalVariable { ordinal: 0 };
+        let mut ordinal = 0;
+        for constant_mir in self.mir.promoted.iter() {
```

Comment:
Is .iter() necessary here? I guess for loops call IntoIter automatically but not Iter?

Reply:
So it seems.